src/fu-engine.c

```diff
@@ -1079,11 +1079,31 @@ static gboolean
 fu_engine_check_requirement_firmware (FuEngine *self, XbNode *req,
 				      FuDevice *device, GError **error)
 {
+	guint64 depth;
+	g_autoptr(FuDevice) device_actual = g_object_ref (device);
 	g_autoptr(GError) error_local = NULL;
 
+	/* look at the parent device */
+	depth = xb_node_get_attr_as_uint (req, "depth");
+	if (depth != G_MAXUINT64) {
+		for (guint64 i = 0; i < depth; i++) {
+			FuDevice *device_tmp = fu_device_get_parent (device_actual);
```

Comment:
I think this might need to be `g_autoptr` and take a reference too since you're assigning it to another `g_autoptr`

Reply:
Nah, `fu_device_get_parent()` is `transfer-none`

Reply:
But then you do a `g_set_object` later, it's still OK?  I'm worried on cleanup you destroy the parent.

Reply:
`g_set_object()` clears the first parameter if non-NULL, then refs the second parameter, saving it into the first. It's like doing:

    if (device_actual != NULL)
        g_object_unref (device_actual);
    device_actual = g_object_ref (device_tmp)

...but as one atomic thing.

Reply:
Ah yeah - thanks for clarifying.